# Случайные события. Вероятность события. Условная вероятность. Формула Байеса

## Задача 1

Подбрасываются три игральных кубика. Оцените вероятность того, что сумма значений этих кубиков не будет превышать 6.
Подсказка: оценкой вероятности может служить, например, относительная частота при большом числе опытов. Попробуйте реализовать наибольшее число опытов, которое позволяют ваш компьютер и/или ваше терпение.

In [71]:
import numpy as np
n=100000000
a=(np.random.randint(1,7,size=n)+np.random.randint(1,7,size=n)+np.random.randint(1,7,size=n))
a

array([12, 10, 12, ...,  8, 12,  9])

Количество сумм значений этих кубиков, которое не будет превышать 6

In [72]:
m=len(a[a<=6])
m

9260013

Относительная частота события:

In [73]:
W=m/n
W

0.09260013

In [74]:
print(f'Вероятность того, что сумма значений этих кубиков не будет превышать 6 равна {W:.4f} или {W*100:.3f}%')

Вероятность того, что сумма значений этих кубиков не будет превышать 6 равна 0.0926 или 9.260%


## Задача 2

Имеется колода в 52 карты. Найти число возможностей вытянуть из неё 4 карты так, чтобы среди них был хотя бы один туз.

In [75]:
from math import factorial
def сombination (n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [76]:
n = сombination(52, 4)
n

270725

Общее число исходов:
$ C_k^n = \frac{n!}{k!(n-k)!}=C_4^{52} = \frac{52!}{4!(52-4)!}=270725$

Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

Количество сочетаний m из 4-х карт хотя бы с одним тузом
$ m=C_1^4 \cdot C_3^{51}= \frac{4!}{1!(4-1)!}\cdot\frac{51!}{3!(51-3)!}=83300$

In [77]:
m = сombination(4, 1) * сombination(51, 3)
print(f'Количество сочетаний m из 4-х карт хотя бы с одним тузом: {m}')

Количество сочетаний m из 4-х карт хотя бы с одним тузом: 83300


In [79]:
p =m/n
print(f'Вероятность вытащить четыре карты хотя бы с одним тузом: {p:.4f} или {p*100:.2f}%')

Вероятность вытащить четыре карты хотя бы с одним тузом: 0.3077 или 30.77%


## Задача 3

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

Общее число исходов:
$ C_k^n = \frac{n!}{k!(n-k)!}=C_3^{60} = \frac{60!}{3!(60-3)!}=34220$

In [80]:
n1 = сombination(60, 3)
n1

34220

Количество исходов того, что среди трёх наугад выбранных вопросов студент знает все

In [81]:
m1 = сombination(50, 3)

In [82]:
p1 =m1/n1
print(f'Вероятность того, что среди трёх наугад выбранных вопросов студент знает все: {p1:.4f} или {p1*100:.2f}%')

Вероятность того, что среди трёх наугад выбранных вопросов студент знает все: 0.5728 или 57.28%


Количество исходов того, что среди трёх наугад выбранных вопросов студент знает (только) два

$ m2=C_2^{50} \cdot C_1^{10}= \frac{50!}{2!(50-2)!}\cdot\frac{10!}{1!(10-1)!}=12250$

In [95]:
m2 = сombination(50, 2)*сombination(10, 1)
m2

12250

In [97]:
p2 =m2/n1
print(f'Вероятность того, что среди трёх наугад выбранных вопросов студент знает (только) два: {p2:.4f} или {p2*100:.2f}%')

Вероятность того, что среди трёх наугад выбранных вопросов студент знает (только) два: 0.3580 или 35.80%


# Задача 4

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

События совместны, может выпасть число больше 3 и это число может быть четным.

И события эти зависимые. Событие АВ - выпало четное число очков, больше 3. Р (АВ) =2/6=1/3.

Если бы события были независимыми, то выполнялась бы равенство. 

Р (АВ) =Р (А) *Р (В). Но это равенство не выполняется так как Р (А) =1/2, Р (В) =1/2. 
$Р (А) \cdot Р (В) =1/2\cdot1/2=1/4$. События зависимые и совместны.

# Задача 5

Допустим, имеется некоторая очень редкая болезнь (поражает p = 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в q = 1% случаев даёт ложный положительный ответ.
Вопрос: какова вероятность, что вы действительно больны ей?
Подсказка: используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

$ P = \frac {P(+|болен)\cdot P(болен)}{P(+|болен)\cdot P(болен)+P(+|здоров)\cdot P(здоров)}$

$P=\frac{(1-q)\cdot p}{(1-q)\cdot p + q\cdot (1-p)}= \frac{p-q \cdot p}{p+q-2\cdot p \cdot q}$

In [90]:
P=(0.001-0.01*0.001)/(0.001+0.01-2*0.001*0.01)
P

0.09016393442622951

In [91]:
print(f'Вероятность что я болен данной болезнью {P*100:.2f}%')

Вероятность что я болен данной болезнью 9.02%
